marimo/_output/formatters/structures.py

```diff
@@ -29,6 +29,10 @@ def _leaf_formatter(value: object) -> bool | None | str | int:
         return f"text/plain+float:{value}"
     if value is None:
         return value
+    if isinstance(value, set):
+        return f"text/plain+set:{str(value)}"
+    if isinstance(value, tuple):
+        return f"text/plain+tuple:{json.dumps(value)}"
```

Comment:
`isinstance(value, tuple)` does not actually get hit because we don't treat tuples as a leaf (which is the right thing probably at the moment). however, we do treat sets as a leaf